rogue-thi-app/components/modal/FilterFoodModal.js

```diff
@@ -0,0 +1,173 @@
+import Modal from 'react-bootstrap/Modal'
+import styles from '../../styles/FilterFoodModal.module.css'
+import Form from 'react-bootstrap/Form'
+import Button from 'react-bootstrap/Button'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { faPen } from '@fortawesome/free-solid-svg-icons'
+import allergenMap from '../../data/allergens.json'
+import flagMap from '../../data/mensa-flags.json'
+import React, { useContext, useState } from 'react'
+import { useFoodFilter } from '../../lib/hooks/food-filter'
+import { ShowFoodFilterModal } from '../../pages/_app'
+
+Object.keys(allergenMap)
+  .filter(key => key.startsWith('_'))
+  .forEach(key => delete allergenMap[key])
+
+/**
+ * A modal component that allows users to personalize their food preferences and allergenes
+ * @returns {JSX.Element}
+ * @constructor
+ */
+export default function FilterFoodModal () {
+  const {
+    selectedRestaurants,
+    preferencesSelection,
+    setPreferencesSelection,
+    allergenSelection,
+    setAllergenSelection,
+    toggleSelectedRestaurant,
+    savePreferencesSelection,
+    saveAllergenSelection
+  } = useFoodFilter()
+  const [showFoodFilterModal, setShowFoodFilterSelection] = useContext(ShowFoodFilterModal)
+  const [showAllergenSelection, setShowAllergenSelection] = useState(false)
+  const [showPreferencesSelection, setShowPreferencesSelection] = useState(false)
+
+  return (
+    <>
+      <Modal show={showFoodFilterModal} onHide={() => setShowFoodFilterSelection(false)}>
+        <Modal.Header closeButton>
+          <Modal.Title>Filter</Modal.Title>
+        </Modal.Header>
+
+        <Modal.Body>
+          <div className={styles.filterHeader}>
+            <h6>
+              Restaurants
+            </h6>
+          </div>
+
+          <div className={styles.filterBody}>
+            <Form.Check
+              id='restaurant-checkbox-mensa'
+              label='Mensa anzeigen'
+              checked={selectedRestaurants.includes('mensa')}
+              onChange={() => toggleSelectedRestaurant('mensa')}
+            />
+            <Form.Check
+              id='restaurant-checkbox-reimanns'
+              label='Reimanns anzeigen'
+              checked={selectedRestaurants.includes('reimanns')}
+              onChange={() => toggleSelectedRestaurant('reimanns')}
+            />
+          </div>
+
+          <hr/>
+
+          <div className={styles.filterHeader}>
+            <h6>
+              Allergien
+            </h6>
+            <Button variant='outline-primary' onClick={() => {
+              setShowAllergenSelection(true)
+              setShowFoodFilterSelection(false)
+            }}>
+              <FontAwesomeIcon title="Allergene" icon={faPen} fixedWidth/>
+            </Button>
+          </div>
+          <div className={styles.filterBody}>
+            <>Ausgewählt:</>
```

Comment:
There should be a space after the colon:
![image](https://user-images.githubusercontent.com/13948008/221993234-c9bd0d8b-a407-4880-913c-fd896c80159d.png)

Reply:
Yes.